src/equations/compressible_euler_2d.jl

```diff
@@ -1588,6 +1588,33 @@ end
     return SVector(w1, w2, w3, w4)
 end
 
+# Transformation from conservative variables u to entropy vector dSdu,
+# S = -rho*s/(gamma-1), s=ln(p)-gamma*ln(rho)
+@inline function cons2entropy_spec(u, equations::CompressibleEulerEquations2D)
+    rho, rho_v1, rho_v2, rho_e = u
+
+    v1 = rho_v1 / rho
+    v2 = rho_v2 / rho
+    v_square = v1^2 + v2^2
+    inv_rho_gammap1 = (1 / rho)^(equations.gamma + 1.0)
+
+    # The derivative vector for the modified specific entropy of Guermond et al.
+    w1 = inv_rho_gammap1 *
+         (0.5 * rho * (equations.gamma + 1.0) * v_square - equations.gamma * rho_e)
+    w2 = -rho_v1 * inv_rho_gammap1
+    w3 = -rho_v2 * inv_rho_gammap1
+    w4 = (1 / rho)^equations.gamma
+
+    # The derivative vector for other specific entropy
+    # sp = 1.0/(gammam1 * (rho_e - 0.5 * rho * v_square)
+    # w1 = gammam1 * 0.5 * v_square * sp - gamma / rho
+    # w2 = -gammam1 * v1 * sp
+    # w3 = -gammam1 * v2 * sp
+    # w4 = gammam1 * sp
```

Comment:
I'm not sure if it makes sense to keep this comment. I copied it from Fluxo back then.
Do you use it sometimes? @amrueda

Reply:
Good question!
We have mainly used this "modified" specific entropy (`s = e * rho^(1 - gamma) = p *rho^(-gamma) / (gamma - 1)`) because it is very well suited for the Newton's method and guarantees that the "conventional" specific entropy (`s = ln(p / rho^gamma)`) of the Euler equations is within bounds. I would definitely add that to the description of the subroutine, as suggested above.

If you are not in the mood for creating new test cases for the "conventional" specific entropy, I think that it is fine if you just remove the comments. 
```suggestion
```

Reply:
Done in [ad7e1da](https://github.com/trixi-framework/Trixi.jl/pull/1792/commits/ad7e1dabf3c532d18b8af45b0b490b47b4a4e3ef)